autoconnect/autoconnect-ws/autoconnect-ws-sm/src/unidentified.rs

```diff
@@ -102,6 +100,58 @@ impl UnidentifiedClient {
         let smsgs = std::iter::once(smsg).chain(check_storage_smsgs);
         Ok((wpclient, smsgs))
     }
+
+    /// Lookup a User or return a new User record if the lookup failed
+    async fn get_or_create_user(&self, uaid: Option<Uuid>) -> DbResult<GetOrCreateUser> {
+        trace!("❓UnidentifiedClient::get_or_create_user");
+        let connected_at = ms_since_epoch();
+
+        if let Some(uaid) = uaid {
+            // NOTE: previously a user would be dropped when
+            // serde_dynamodb::from_hashmap failed (but this now occurs inside
+            // the db impl)
+            let maybe_user = self.app_state.db.get_user(&uaid).await?;
+            if let Some(mut user) = maybe_user {
+                let maybe_flags = process_existing_user(&self.app_state, &user).await?;
+                if let Some(flags) = maybe_flags {
+                    user.node_id = Some(self.app_state.router_url.to_owned());
+                    user.connected_at = connected_at;
+                    user.set_last_connect();
+                    self.app_state.db.update_user(&user).await?;
+                    return Ok(GetOrCreateUser {
+                        user,
+                        existing_user: true,
+                        flags,
+                    });
+                }
+            }
+            // NOTE: when the client's specified a uaid but get_user returns
```

Comment:
Wanted to point this out as one difference between the other state machine. Previously we only set the `defer_add_user` (previously called `deferred_registration`) for when clients did not specify a uaid.

Now we're doing it for any case where we would have created a brand new user record -- including when a user specifies a `uaid` and its lookup failed or its record was invalid.